test/test_automl/test_automl.py

```diff
@@ -114,6 +115,33 @@ def test_fit(self):
         self._tearDown(backend_api.temporary_directory)
         self._tearDown(backend_api.output_directory)
 
+    def test_delete_non_winning_models(self):
+        backend_api = self._create_backend(
+            'test_delete', delete_tmp_folder_after_terminate=False)
+
+        X_train, Y_train, _, _ = putil.get_dataset('iris')
+        automl = autosklearn.automl.AutoML(
+            backend_api,
+            time_left_for_this_task=20,
+            per_run_time_limit=5,
+            ensemble_nbest=3,
+        )
+        automl.fit(
+            X_train, Y_train, metric=accuracy, task=MULTICLASS_CLASSIFICATION,
+        )
+
+        model_files = glob.glob(os.path.join(
+            backend_api.temporary_directory, '.auto-sklearn', 'models', '*.model'))
```

Comment:
Isn't there a function in the backend to get the relevant model identifiers?

Reply:
There is indeed: list_all_models()
Let's use it then.